### DISCUSSION TITLE
Am I supposed decode the payload string myself?

### DISCUSSION THREAD

**Discussion by HubKing** [dotnet/MQTTnet#1782] [Q&A] [answered]
I can publish a string easily with `WithPayload("Working")`, but how am I supposed to get that string in the subscriber? I could manually do it by

    //arg is MqttApplicationMessageReceivedEventArgs
    var msg = Encoding.UTF8.GetString(args.ApplicationMessage.PayloadSegment);

but am I not sure if this is the recommended way. Looks too clunky. I saw the example, and it was only using `DumpToConsole` or something. Isn't there anything like `args.PayloadString`?

**Comment by CZEMacLeod** [comment, marked as answer]
@HubKing I've done a simple example available at https://github.com/CZEMacLeod/Tasmota-Mosquitto-MQTTnet-Sample
There is an extension method `ConvertPayloadToString` which may do what you want, but as many payloads are utf8 encoded numbers, or often utf8 json payloads, converting them to Strings is a lot of memory usage and copying.

I've shown number parsing using `Utf8Parser` and Json decoding using `System.Text.Json`.

There are a couple of other little nice example bits in there too. Hope this helps.
